mr/src/main/java/org/apache/iceberg/mr/InputFormatConfig.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.mr;
+
+import java.util.function.Function;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.SchemaParser;
+import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.expressions.Expression;
+
+public class InputFormatConfig {
```

Comment:
This is from #933, specifically at https://github.com/apache/iceberg/pull/933/files#diff-0aa47c0ca21b715226d83ae55f4b4671 so we just pulled that class in here "as is" as it was easier. If you'd prefer we can trim this down to just the config needed by the SerDe or we can (temporarily) put the code elsewhere in this PR and then refactor it back into this class when we come back to#933 after this is merged.

Reply:
I think it's safer to either reference it in place or to move it in this commit. Otherwise, changes to the other class may not show up as git commit conflicts and we could accidentally lose them by adding the class in one commit and removing the other copy later.

Reply:
I don't think I understand what you want us to do here. We need it here for this to compile and get merged in separately from #933. We basically cut and paste the class out of the other branch to here. We'll happily resolve any conflicts etc. in #933 later if needed.

Reply:
If we are moving code into its own top-level class, then we need to delete the code that is moving from its original location so we don't drop changes that are introduced in the mean time.

Reply:
Ah, OK, now I see what you mean, yes, that makes sense. I've done that [now](https://github.com/apache/iceberg/pull/1103/commits/2350025f11bf3ce6df226d21a87d5c403ab50b53)